polyply/src/topology.py

```diff
@@ -388,7 +388,8 @@ def add_positions_from_file(self, path, skip_res=[], resolution='mol'):
         for meta_mol in self.molecules:
             for meta_node in meta_mol.nodes:
                 resname = meta_mol.nodes[meta_node]["resname"]
-                mol_nodes = meta_mol.nodes[meta_node]['graph'].nodes
+                # the fragment graph nodes are not sorted
+                mol_nodes = sorted(meta_mol.nodes[meta_node]['graph'].nodes)
```

Comment:
Sort by node idx or one of the node attributes (e.g. 'atomidx')? This puts quite some semantics on node indices which are otherwise meaningless.

Reply:
within polyply all the node indices are strictly related to atom indices by `node_idx = atom_idx-1`. That is strictly enforced by the itp-reader, but I see the point here. Perhaps better to sort by atom-index or whatever it is called.

Reply:
fixed it